workflow/functional/structure_map.py

```diff
@@ -0,0 +1,11 @@
+
+
+def structure_map(fn, x):
```

Comment:
usually called `fmap` short for `functor map` https://wiki.haskell.org/Functor

Reply:
`fmap` seems to work on both key and value though

Reply:
no it applies over the inner type

Reply:
Look at the hashmap for example:

https://hackage.haskell.org/package/hashmap-1.3.3/docs/Data-HashMap.html

only `Map[K, .]` (i.e. `Map k` in the docs) is a functor, which is the "natural" choice

you can of course construct a reverse map and have the keys as the natural thing to map over :P

Reply:
it's perhaps a bit silly to refer to Haskell docs, but if they got something right its the way they type stuff

Reply:
I don't know haskell but I interpreted this as f on both the key and value?
`fmap :: (a -> b) -> f a -> f b`

Reply:
perhaps its better to have something without the connotation that fmap brings :)